### DISCUSSION TITLE
What Zope Interface is used for?

### DISCUSSION THREAD

**Discussion by eisenhaus335** [Pylons/pyramid#3756] [General]
I have read many documentation explanation and read the code repo for Zope. I read the Interfaces and how it integrates for example transaction packages. But how it should be use for example in maintaining the code? I don't follow the history of Pyramid and Zope so its really out of world way to code python web projects

I was looking for sqlalchemy db for writing http interface for ArcadeDB (Which use http interface). I look pyramid-mailer for example although i could understand some of it. I am confused with the Zope Interface cause it looks important and at the sametime seems odd. Maybe if some maintainer of Pyramid could give some history lesson and so haha

**Comment by mmerickel** [comment]
Have you had a chance to read the [design defense](https://docs.pylonsproject.org/projects/pyramid/en/latest/designdefense.html)? It has some details in there about Pyramid's usage of Zope. It's important to note that you can use Pyramid without knowing anything about Zope - it's just an implementation detail unless you choose to get deeper into it.
